### PR TITLE
[TESTING] Localisation reset function

### PR DESCRIPTION
**Pull request by danielginn**
Allows new particles to reset the current particles and calculates Htw for each self particle.

### CODE REVIEW

File: module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp [hunk 1]
```diff
@@ -123,6 +129,36 @@ namespace localisation {
...
+                const Transform3D& Htw = convert<double, 4, 4>(sensors.world);
+                arma::vec3 rTFf;
+                arma::mat22 Hfw_xy;
+                arma::mat22 pos_cov;
+                arma::mat33 state_cov;
+                std::vector<arma::vec3> states;
+                std::vector<arma::mat33> cov;
+
+                for (auto& s : locReset.hypotheses) {
+                    rTFf              = {s.position[0], s.position[1], 0};
```
Comment (JakeFountain) [inline module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp]: initialise these variables in the for loop. cpp is smart enough to optimise and not re-create it every loop. this also removes the possibility of using values from the last loop

File: module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp [hunk 2]
```diff
@@ -123,6 +129,36 @@ namespace localisation {
...
+                    Hft.translation() = -rTFf;
```
Comment (JakeFountain) [inline module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp]: why negate here?

File: module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp [hunk 3]
```diff
@@ -123,6 +129,36 @@ namespace localisation {
...
+                    Hft.rotateZ(s.heading);
+                    Hfw = Hft * Htw;
```
Comment (JakeFountain) [inline module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp]: need to project to transform2d

File: shared/utility/localisation/transform.h [hunk 4]
```diff
@@ -34,6 +34,16 @@ namespace localisation {
...
     }
 
+    // Transforms the transform
+    inline arma::vec3 transform3DToFieldState(const utility::math::matrix::Transform3D& m) {
+        utility::math::matrix::AxisAngle ax = m.rotation().axisAngle();
+        if (!(arma::approx_equal(ax.first, arma::vec3({0, 0, 1}), "absdiff", 0.00001)
+              || arma::approx_equal(ax.first, arma::vec3({0, 0, -1}), "absdiff", 0.00001))) {
+            throw std::runtime_error("transform.h MatrixToLocalisationState - matrix must be a z rotation");
+        }
+        return arma::vec3({m.translation()[0], m.translation()[1], ax.second});
```
Comment (JakeFountain) [inline shared/utility/localisation/transform.h]: negate if axis backwards

File: module/GameControllerSimulator/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+GameControllerSimulator
```
Comment (TrentHouliston) [inline module/GameControllerSimulator/README.md]: This shouldn't be in this PR

File: module/vision/GoalDetector/src/GoalDetector.cpp [hunk 6]
```diff
@@ -557,6 +557,7 @@ namespace vision {
                         Goal::MeasurementType::LEFT_NORMAL, left, left_vecCov, left_Angles, left_AngCov));
 
                     // TR BR cross product gives right side
+
```
Comment (TrentHouliston) [inline module/vision/GoalDetector/src/GoalDetector.cpp]: Delete these for conflicts sake!

File: shared/utility/math/filter/ParticleFilter.h [hunk 7]
```diff
@@ -91,11 +91,13 @@ namespace math {
             ParticleList getParticles(const StateVec& initialMean,
                                       const StateMat& initialCovariance,
                                       const int& n_particles) const {
+
```
Comment (TrentHouliston) [inline shared/utility/math/filter/ParticleFilter.h]: Delete these to help conflicts

**Pull request merged**